### PR TITLE
Feature/disable pattern validation

### PR DESCRIPTION
**Pull request by cmfsotelo**
Despite the name of the branch, this developments aim to put the behaviour of checking for the pattern optional, not removing it.

Added a preference **CheckPattern** that, when set to false, disables the pattern checking behaviour.

Without setting the preference, everything stays the same.

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version='1.0' encoding='utf-8'?>
-<plugin id="outsystems-secure-device" version="2.0.0.os" xmlns="http://apache.org/cordova/ns/plugins/1.0" xmlns:android="http://schemas.android.com/apk/res/android">
+<plugin id="outsystems-secure-device" version="2.0.1" xmlns="http://apache.org/cordova/ns/plugins/1.0" xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (Chuckytuh) [inline plugin.xml]: If this PR isn't bringing the entire code from the original repo, I'd rather maintain the version format major.minor.patch-OS (2.0.1-OS), this way we can easily track changes from the original repository

**Review by Chuckytuh — CHANGES_REQUESTED** [review]

File: plugin.xml (hunk 1 above)
Comment (cmfsotelo) [reply plugin.xml]: Ok, that makes sense.
However, some issues may arise depending of the approaches:

- If we keep the 2.0.0.os, we loose track of our versions
- If we change to 2.0.1.os we have a versioning issue if the original repo releases a 2.0.1 version and we merge those developments unto ours.
 what is your suggestion for the version name?

Comment (Chuckytuh) [reply plugin.xml]: Heads up: format `major.minor.patch.OS` is not a valid semver format and will cause problems when installing using npm, instead, make sure it's set as `major.minor.patch-OS`.

Assuming that the original repository is on version `1.0.4` and this repository has changes over the original repository, we would have `1.0.4-OS` for the first iteration of changes. For each new bug fix/feature introduced in this fork, the suffixed value on `OS` is increased, for example: `1.0.4-OS1` and then `1.0.4-OS2`, etc.. 
This way we can keep track from which version the repository originated.

Now imagine that if the upstream increased to a new major (2.0.0), we would ideally merge from the upstream into this fork and set it as 2.0.0-OS, resetting the suffix value.

Cheers

**Review by cmfsotelo — COMMENTED** [review]
Updated version number to 2.0.0-os2

File: plugin.xml (hunk 1 above)
Comment (cmfsotelo) [reply plugin.xml]: perfect
Resolved ;)

**Review by Chuckytuh — APPROVED** [review]

**Pull request merged**